torch/_numpy/_funcs_impl.py

```diff
@@ -359,12 +359,13 @@ def arange(
 
     # the dtype of the result
     if dtype is None:
-        dtype = _dtypes_impl.default_dtypes().int_dtype
-    # XXX: default values do not get normalized
-    start, stop, step = (_util._coerce_to_tensor(x) for x in (start, stop, step))
-
-    dummy = torch.empty(1, dtype=dtype)
-    target_dtype = _dtypes_impl.result_type_impl(start, stop, step, dummy)
+        target_dtype = (
+            _dtypes_impl.default_dtypes().float_dtype
+            if any(_dtypes_impl.is_float_or_fp_tensor(x) for x in (start, stop, step))
+            else _dtypes_impl.default_dtypes().int_dtype
+        )
+    else:
+        target_dtype = dtype
 
     # work around RuntimeError: "arange_cpu" not implemented for 'ComplexFloat'
     work_dtype = torch.float64 if target_dtype.is_complex else target_dtype
```

Comment:
You said the target dtype cannot be complex?

Reply:
It can be complex. `np.arange(1+1j, 2+2j)` works. The code above should be adapted to find whether it's int, float or complex.

Reply:
They are documented as `start: integer or real, optional` --- https://numpy.org/doc/stable/reference/generated/numpy.arange.html

That complex arguments are accepted at all appears to be implementation details leaking, with rather weird and inconsistent results:

```
In [1]: import numpy as np

In [2]: np.arange(1 + 1j, 5+5j, 1)
Out[2]: array([1.+1.j, 2.+1.j, 3.+1.j, 4.+1.j])

In [3]: np.arange(1 + 1j, 5+5j, 1+1j)
Out[3]: array([], dtype=complex128)

In [4]: np.arange(1 + 1j, 5+5j, 1j)
Out[4]: array([], dtype=complex128)
```

The last commit just falls back to eager.